src/cirrus/plugins/deployers/chef.py

```diff
@@ -44,6 +59,110 @@ def deploy(self, opts):
         LOGGER.info('Chef deployment running...')
         LOGGER.info(opts)
 
+        args = self._read_cirrus_conf()
+        for opt in vars(opts):
+            if (opt in args) and (getattr(opts, arg, None) is not None):
```

Comment:
This will ignore an arg that has been deliberately set to None. That's probably okay, though ...